TUTORIAL.md

```diff
@@ -0,0 +1,245 @@
+#Guide to Solving and Reviewing Rails Blogs Nested Forms
+##Overview
+This lab will teach you how to implement nested forms. In this example, we will nest tags under posts, so a user can submit a post that has many tags.
+
+###Step 1: Set up your app
+- `bin/rake db:migrate RAILS_ENV=test`
+- `bundle install`
+- `rails s` then visit `localhost:3000`
+- `rake db:seed`
+
+At this point we should have a form that will let us create a post, along with tags that we can select. Now we need a way to add a new tag, so let's add that to our form.
+
+###Step 2: Use fields_for to accept tags
+
+```ruby
+<div class="field">
+  <%= f.fields_for(:tags) do |tag_form| %>
+    <%= tag_form.label :name %>
+    <%= tag_form.text_field :name %>
+  <% end %>
+</div>
+
+```
+####Completed Form
+
+```ruby
+<%= form_for(@post) do |f| %>
+  <% if @post.errors.any? %>
+    <div id="error_explanation">
+      <h2><%= pluralize(@post.errors.count, "error") %> prohibited this post from being saved:</h2>
+      <ul>
+      <% @post.errors.full_messages.each do |msg| %>
+        <li><%= msg %></li>
+      <% end %>
+      </ul>
+    </div>
+  <% end %>
+
+  <div class="field">
+    <%= f.label :name %><br>
+    <%= f.text_field :name %><br>
+    <%= f.label :content %><br>
+    <%= f.text_area :content %><br>
+    Tags:
+  </div>
+
+  <div class="field">
+    <%= f.fields_for(:tags) do |tag_form| %>
+      <%= tag_form.label :name %>
+      <%= tag_form.text_field :name %>
+    <% end %>
+  </div>
+
+  <div class="field">
+    <%= f.collection_check_boxes :tag_ids, Tag.all, :id, :name %><br>
+  </div>
+
+  <div class="actions">
+    <%= f.submit %>
+  </div>
+
+<% end %>
+```
+
+####Before we go any further, let's break down what is happening above. 
+
+Our `form_for` accepts `@post` as an argument, which is an instance of our `Post` class. We then pass it to the variable `f` to build our form.
```

Comment:
So actually what is going on here is you are passing the `@post` variable to the `form_for` method. This method also takes in a block with a single argument to the block. That argument is a `FormBuilder` object. Take a look at [this documentation](http://api.rubyonrails.org/classes/ActionView/Helpers/FormHelper.html#method-i-form_for) for a bit more. The FormBuilder object that is yielded to the block then contains all the information about the model that is passed in, and has a bunch of [instance methods](http://api.rubyonrails.org/classes/ActionView/Helpers/FormBuilder.html)